src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -169,4 +179,169 @@ public static String composeClassHeaderEndpointSampleCode(
+              .setExprReferenceExpr(clientVarExpr)
+              .setMethodName(JavaStyle.toLowerCamelCase(method.name()))
+              .setArguments(methodArgExprs)
+              .setReturnType(clientType)
+              .build());
+    } else {
+      VariableExpr responseVarExpr =
+          VariableExpr.withVariable(
+              Variable.builder().setName(RESPONSE).setType(method.outputType()).build());
+      bodyExpr.add(
+          createAssignExprForVariableWithClientMethod(
+              responseVarExpr,
+              clientVarExpr,
+              JavaStyle.toLowerCamelCase(method.name()),
+              methodArgExprs));
+    }
+
+    return TryCatchStatement.builder()
+        .setTryResourceExpr(assignClientVariableWithCreateMethodExpr(clientVarExpr))
+        .setTryBody(
+            bodyExpr.stream().map(e -> ExprStatement.withExpr(e)).collect(Collectors.toList()))
+        .setIsSampleCode(true)
+        .build();
+  }
+
+  // ==================================Helpers===================================================//
+
+  // Assign client variable expr with create client.
+  // e.g EchoClient echoClient = EchoClient.create()
+  private static AssignmentExpr assignClientVariableWithCreateMethodExpr(
+      VariableExpr clientVarExpr) {
+    return AssignmentExpr.builder()
+        .setVariableExpr(clientVarExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(
+            MethodInvocationExpr.builder()
+                .setStaticReferenceType(clientVarExpr.variable().type())
+                .setReturnType(clientVarExpr.variable().type())
+                .setMethodName("create")
+                .build())
+        .build();
+  }
+
+  // Create a list of Expr for method argument variable expression.
+  private static List<Expr> createMethodArgVarExprs(
+      List<MethodArgument> arguments, Map<String, ResourceName> resourceNames) {
+    return arguments.stream()
+        .map(
+            arg -> {
+              VariableExpr argVarExpr =
+                  VariableExpr.withVariable(
+                      Variable.builder()
+                          .setName(JavaStyle.toLowerCamelCase(arg.name()))
+                          .setType(arg.type())
+                          .build());
+              if (arg.type().equals(TypeNode.STRING)
+                  && arg.field().hasResourceReference()
+                  && resourceNames.containsKey(
+                      arg.field().resourceReference().resourceTypeString())) {
+                return MethodInvocationExpr.builder()
+                    .setExprReferenceExpr(argVarExpr)
+                    .setMethodName("toString")
+                    .build();
+              }
+              return argVarExpr;
+            })
+        .collect(Collectors.toList());
+  }
+
+  // Return a list of AssignmentExpr for method argument with its default value.
+  private static List<Expr> assignMethodArgumentsWithDefaultValues(
+      List<MethodArgument> arguments,
```

Comment:
We should avoid this kind of coupling (i.e. `arguments` and `argVarExprs`) where possible. Instead, process the list of `argVarExprs`  (making any conversions separately) before calling this method, then we won't need to pass in `arguments` at all.